src/main/java/dal/fatih/todoly/TaskRepository.java

```diff
@@ -0,0 +1,165 @@
+package dal.fatih.todoly;
+
+
+import java.io.Closeable;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TaskRepository implements Closeable {
+
+    private final DBConnection dbConnection = new DBConnection();
+    private final Connection connection = dbConnection.getConnection();
+    private final PreparedStatement createPrepareStatement;
+    private final PreparedStatement listPrepareStatement;
+    private final PreparedStatement getPreparedStatement;
+    private final PreparedStatement deleteByIdPreparedStatement;
+    private final PreparedStatement filterPreparedStatement;
+    private final PreparedStatement getByTitleOrDesPreparedStatement;
+
+
+    public TaskRepository() throws SQLException {
+
+        createTable();
+        createPrepareStatement =
+                connection.prepareStatement("insert into tasks " + "values (?,?,?,?,?)");
+        listPrepareStatement =
+                connection.prepareStatement("select id,taskid,title,description,duedate from tasks");
+        getPreparedStatement =
+                connection.prepareStatement("select taskid,title,description,duedate from tasks where taskid=? limit 1");
+        deleteByIdPreparedStatement =
+                connection.prepareStatement("delete from tasks where taskid=?");
+        filterPreparedStatement =
+                connection.prepareStatement("select * from tasks where duedate between now() and ?");
+        getByTitleOrDesPreparedStatement =
+                connection.prepareStatement("select taskid,title,description,duedate from tasks where title like ? "
+                        + "or description like ?");
+    }
+
+    public void createTable() {
+        try {
+            PreparedStatement preparedStatement = connection.prepareStatement("create table if not exists tasks"
+                    + "  (id int auto_increment primary key not null,"
+                    + "   taskid uuid(36),"
+                    + "   title varchar(35),"
+                    + "   description varchar(250),"
+                    + "   duedate date)");
+            preparedStatement.executeUpdate();
+            preparedStatement.close();

```

Comment:
Don't use prepared statements everywhere. It is for making repeatedly running SQL queries efficient. Create table is executed only once, so it can be executed as a regular statement.
```suggestion
            Statement statement = connection.createStatement();
            statement.execute("create table if not exists tasks"
                    + "  (id int auto_increment primary key not null,"
                    + "   taskid uuid(36),"
                    + "   title varchar(35),"
                    + "   description varchar(250),"
                    + "   duedate date)");
            statement.close();
```